modules/renter/contractor/contracts.go

```diff
@@ -83,6 +84,104 @@ func (c *Contractor) contractEndHeight() types.BlockHeight {
 	return endHeight
 }
 
+// managedMarkContractsUtility checks every active contract in the contractor and
+// figures out whether the contract is useful for uploading, and whehter the
+// contract should be renewed.
+func (c *Contractor) managedMarkContractsUtility() {
+	// Pull a new set of hosts from the hostdb that could be used as a new set
+	// to match the allowance. The lowest scoring host of these new hosts will
+	// be used as a baseline for determining whether our existing contracts are
+	// worthwhile.
+	c.mu.RLock()
+	hostCount := int(c.allowance.Hosts)
+	c.mu.RUnlock()
+	if hostCount <= 0 {
+		return
+	}
+	hosts := c.hdb.RandomHosts(hostCount+minScoreHostBuffer, nil)
+
+	// Find the lowest score of this batch of hosts.
+	lowestScore := c.hdb.ScoreBreakdown(hosts[0]).Score
```

Comment:
This nondeterministically causes a `index out of range` panic on my machine. I think we need a `if len(hosts)` branch before this line